ray-operator/controllers/ray/rayservice_controller.go

```diff
@@ -324,7 +323,7 @@ func (r *RayServiceReconciler) shouldPrepareNewRayCluster(rayServiceInstance *ra
 		// Prepare new RayCluster if:
 		// 1. No active and pending cluster
 		// 2. No pending cluster, and the active RayCluster has changed.
-		if activeRayCluster == nil || !reflect.DeepEqual(activeRayCluster.Spec, rayServiceInstance.Spec.RayClusterSpec) {
+		if activeRayCluster == nil || !utils.CompareJsonStruct(activeRayCluster.Spec, rayServiceInstance.Spec.RayClusterSpec) {
```

Comment:
Could you explain this diff?
I guess you could have different objects that serialize to the same JSON and this prevents you from considering them different.
Should we always be using this method in place of DeepEqual?

Reply:
Suggest to use this for json struct comparison. The deepEqual will fail in some cases.